lib/ex_doc/retriever.ex

```diff
@@ -379,16 +379,26 @@ defmodule ExDoc.Retriever do
 
   defp get_callback(callback, source, optional_callbacks, module_data) do
     {:docs_v1, _, _, content_type, _, _, _} = module_data.docs
-    {{kind, name, arity}, anno, _, doc, metadata} = callback
+    {{kind, name, arity}, anno, signature, doc, metadata} = callback
     actual_def = actual_def(name, arity, kind)
     doc_line = anno_line(anno)
-    annotations = annotations_from_metadata(metadata)
+    signature = if signature == [], do: nil, else: Enum.join(signature, " ")
```

Comment:
@josevalim I noticed that we were ignore signature from the chunk entry for both callbacks and types so I thought I'd default to it. Was there a particular reason we didn't do it before? If this is the way to go, I'm happy to make sure Elixir is emitting the signature in the chunk.

Reply:
It is on purpose to keep the chunks lean. If there is no signature, we fetch it from the AST. I would keep it as is for now.